src/main/native/cpp/view/AnalyzerPlot.cpp

```diff
@@ -248,31 +275,73 @@ void AnalyzerPlot::DisplayTimeDomainPlots() {
     const char* y =
         i % 2 == 0 ? "Velocity (units / s)" : "Acceleration (units / s / s)";
 
+    // Get a reference to the data we are plotting.
+    auto& filteredData = m_filteredData[kChartTitles[i]];
+    auto& rawData = m_rawData[kChartTitles[i]];
+
+    // Generate Sim vs Filtered Plot
     if (m_fitNextPlot[i]) {
       ImPlot::FitNextPlotAxes();
     }
     if (ImPlot::BeginPlot(kChartTitles[i], x, y, ImVec2(-1, 0),
                           ImPlotFlags_None, ImPlotAxisFlags_NoGridLines)) {
-      // Get a reference to the data we are plotting.
-      auto& data = m_data[kChartTitles[i]];
+      // Set Legend Location:
+      ImPlot::SetLegendLocation(ImPlotLocation_East, ImPlotOrientation_Vertical,
+                                true);
+
+      // Plot simulated time-domain data or raw data.
+      if (i == 2 || (i == 4)) {
```

Comment:
This whole if structure could use refactoring for readability and reduced duplication.